src/client.ts

```diff
@@ -461,164 +363,39 @@ export default class Client implements ClientInterface {
-
-        return this.bundleInstall(customGemfilePath, {
-          progress,
-          token: this.bundleInstallCancellationSource.token,
-        });
-      }
+  private async installOrUpdateServer(): Promise<void> {
+    const oneDayInMs = 24 * 60 * 60 * 1000;
+    const lastUpdatedAt: number | undefined = this.context.workspaceState.get(
+      "rubyLsp.lastGemUpdate"
     );
 
-    // Update the last time we checked for updates
-    this.context.workspaceState.update("rubyLsp.lastBundleInstall", Date.now());
-  }
-
-  private async bundleInstall(
-    bundleGemfile?: string,
-    options?: {
-      progress?: vscode.Progress<{ message?: string }>;
-      token?: vscode.CancellationToken;
-    }
-  ) {
-    const env = { ...this.ruby.env, BUNDLE_GEMFILE: bundleGemfile };
-
-    return new Promise<void>((resolve, reject) => {
-      const abortController = new AbortController();
-
-      if (options?.token) {
-        options.token.onCancellationRequested(() => {
-          abortController.abort();
-        });
-      }
-
-      this.outputChannel.appendLine(">> Running `bundle install`...");
+    const { stdout } = await asyncExec("gem list ruby-lsp", {
+      cwd: this.workingFolder,
+      env: this.ruby.env,
+    });
 
-      const install = spawn("bundle", ["install"], {
-        shell: true,
+    // If the gem is not yet installed, install it
+    if (!stdout.includes("ruby-lsp")) {
+      await asyncExec("gem install ruby-lsp", {
         cwd: this.workingFolder,
-        signal: abortController.signal,
-        env,
+        env: this.ruby.env,
       });
 
-      const stdout = readline
-        .createInterface({
-          input: install.stdout,
-          terminal: false,
-        })
-        .on("line", (line) => {
-          this.outputChannel.appendLine(`BUNDLER> ${line}`);
-
-          if (options?.progress) {
-            options.progress.report({ message: line });
-          }
-        });
-
-      const stderr = readline
-        .createInterface({
-          input: install.stderr,
-          terminal: false,
-        })
-        .on("line", (line) => {
-          this.outputChannel.appendLine(`ERROR> ${line}`);
-        });
-
-      install.on("close", (code) => {
-        stdout.close();
-        stderr.close();
-
-        this.outputChannel.appendLine(
-          `>> \`bundle install\` exited with code ${code}`
-        );
-
-        if (code === 0) {
-          resolve();
-        } else {
-          reject(new Error(`bundle install exited with code ${code}`));
-        }
-      });
+      this.context.workspaceState.update("rubyLsp.lastGemUpdate", Date.now());
+      return;
+    }
 
-      install.on("error", (error) => {
-        stdout.close();
-        stderr.close();
-        reject(error);
+    // If we haven't updated the gem in the last 24 hours, update it
+    if (
+      lastUpdatedAt === undefined ||
+      Date.now() - lastUpdatedAt > oneDayInMs
+    ) {
+      await asyncExec("gem update ruby-lsp", {
```

Comment:
Can we have some fault-tolerant mechanism here so it doesn't break when the user has no internet connection?

Reply:
Unfortunately the code I got was `1` 😅 

```
$ gem update ruby-lsp
ERROR:  While executing gem ... (Gem::RemoteFetcher::FetchError)
    SocketError: Failed to open TCP connection to rubygems.org:443 (https://rubygems.org/specs.4.8.gz) (Gem::RemoteFetcher::FetchError)
        /opt/rubies/3.2.2/lib/ruby/3.2.0/rubygems/remote_fetcher.rb:261:in `rescue in fetch_path'
        /opt/rubies/3.2.2/lib/ruby/3.2.0/rubygems/remote_fetcher.rb:241:in `fetch_path'
...

$ echo $?            
1
```

Reply:
Well, thinking more about it, we don't want to prevent the server from starting on any `gem update` errors. I added a `try catch` block to log when something goes wrong.